src/parse-proxy-response.ts

```diff
@@ -34,10 +34,12 @@ export default function parseProxyResponse(
 
 		function onclose(err?: Error) {
 			debug('onclose had error %o', err);
+			reject(err);
 		}
 
 		function onend() {
 			debug('onend');
+			reject(null);
```

Comment:
I think it would be best to throw an actual Error.
No error would also break compatibility with node-fetch.
```suggestion
			reject(new Error('Socket was closed by other party!'));
```

Reply:
Suggestion applied.

Reply:
> Suggestion applied.